### PR TITLE
Add new user_data API helpers (#150)

### PR DESCRIPTION
**Pull request by moul**

### CODE REVIEW

**Comment by aimxhaisse** [comment]
lgtm

File: pkg/api/api.go [hunk 1]
```diff
@@ -1322,6 +1323,98 @@ func (s *ScalewayAPI) GetBootscript(bootscriptID string) (*ScalewayBootscript, e
...
+	var err error
+	resp, err := s.GetResponse("servers/" + serverID + "/user_data/" + key)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("no such user_data %q (%d)", key, resp.StatusCode)
+	}
+
+	defer resp.Body.Close()
```
Comment (QuentinPerez) [inline pkg/api/api.go]: This line should not be after the line 1359 ?

File: pkg/api/api.go [hunk 2]
```diff
@@ -1322,6 +1323,98 @@ func (s *ScalewayAPI) GetBootscript(bootscriptID string) (*ScalewayBootscript, e
...
+	return fmt.Errorf("cannot set user_data (%d)", resp.StatusCode)
+}
+
+// DeleteUserdata deletes a server user_data
+func (s *ScalewayAPI) DeleteUserdata(serverID string, key string) error {
+	resp, err := s.DeleteResponse(fmt.Sprintf("servers/%s/user_data/%s", serverID, key))
+	if err != nil {
+		return err
+	}
+
```
Comment (QuentinPerez) [inline pkg/api/api.go]: defer resp.Body.Close()

File: pkg/cli/x_userdata.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		key := parts[0]
+		switch len(parts) {
+		case 1:
+			// Get userdatas
+			res, err := ctx.API.GetUserdata(serverID, key)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(ctx.Stdout, "%s\n", res.String())
+		default:
```
Comment (QuentinPerez) [inline pkg/cli/x_userdata.go]: case 2:

File: pkg/api/api.go (hunk 1 above)
Comment (moul) [inline pkg/api/api.go]: yep :)

**Comment by moul** [comment]
Fixed

**Pull request merged**